plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevControllerTest.kt

```diff
@@ -96,12 +153,248 @@ class FeatureDevControllerTest : FeatureDevTestBase() {
+
+        val mockInteraction = mock<Interaction>()
+
+        val mockSession = mock<Session>()
+        whenever(mockSession.send(userMessage)).thenReturn(mockInteraction)
+        whenever(mockSession.sessionState).thenReturn(
+            PrepareCodeGenerationState(
+                testTabId, "", mock(), filePaths, deletedFiles, references, testUploadId, 0, messenger
+            )
+        )
+        whenever(mockSession.retries).thenReturn(3)
+
+        controller.onCodeGeneration(mockSession, userMessage, testTabId)
+
+        coVerifyOrder {
+            messenger.sendAnswer(testTabId, message("amazonqFeatureDev.code_generation.no_file_changes"), FeatureDevMessageType.Answer)
+            messenger.sendSystemPrompt(
+                testTabId,
+                listOf(FollowUp(FollowUpTypes.RETRY, message("amazonqFeatureDev.follow_up.retry"), status = FollowUpStatusType.Warning))
+            )
+            messenger.sendChatInputEnabledMessage(testTabId, false)
+        }
+    }
+
+    @Test
+    fun `test handleChat onCodeGeneration doesn't return any files no retries`() = runTest {
+        val filePaths = emptyList<NewFileZipInfo>()
+        val deletedFiles = emptyList<DeletedFileInfo>()
+        val references = listOf(CodeReference(information = "test"))
+
+        val mockInteraction = mock<Interaction>()
+
+        val mockSession = mock<Session>()
+        whenever(mockSession.send(userMessage)).thenReturn(mockInteraction)
+        whenever(mockSession.sessionState).thenReturn(
+            PrepareCodeGenerationState(
+                testTabId, "", mock(), filePaths, deletedFiles, references, testUploadId, 0, messenger
+            )
+        )
+        whenever(mockSession.retries).thenReturn(0)
+
+        controller.onCodeGeneration(mockSession, userMessage, testTabId)
+
+        coVerifyOrder {
+            messenger.sendAnswer(testTabId, message("amazonqFeatureDev.code_generation.no_file_changes"), FeatureDevMessageType.Answer)
+            messenger.sendSystemPrompt(testTabId, emptyList())
+            messenger.sendChatInputEnabledMessage(testTabId, false)
+        }
+    }
+
+    @Test
+    fun `test processFileClicked changes the state of the clicked file`() = runTest {
+        val message = IncomingFeatureDevMessage.FileClicked(testTabId, newFileContents[0].zipFilePath, "", "")
+
+        whenever(featureDevClient.createTaskAssistConversation()).thenReturn(exampleCreateTaskAssistConversationResponse)
+        whenever(chatSessionStorage.getSession(any(), any())).thenReturn(spySession)
+        whenever(spySession.sessionState).thenReturn(
+            PrepareCodeGenerationState(
+                testTabId, "", mock(), newFileContents, deletedFiles, references, testUploadId, 0, messenger
+            )
+        )
+
+        controller.processFileClicked(message)
+
+        val newFileContentsCopy = newFileContents.toList()
+        newFileContentsCopy[0].rejected = !newFileContentsCopy[0].rejected
```

Comment:
If this is expected to be tested use `assert`.

Reply:
I'm doing that in the verify below, I just needed to get the element that should have been rejected from the original list used as input